inference-engine/src/transformations/src/ngraph_ops/gru_sequence_ie.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph_ops/gru_sequence_ie.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::GRUSequenceIE::type_info;
+
+op::GRUSequenceIE::GRUSequenceIE(const Output<Node>& X,
+                                 const Output<Node>& H_t,
+                                 const Output<Node>& WR,
+                                 const Output<Node>& B,
+                                 std::size_t hidden_size,
+                                 op::RecurrentSequenceDirection direction,
+                                 const std::vector<std::string>& activations,
+                                 const std::vector<float>& activations_alpha,
+                                 const std::vector<float>& activations_beta,
+                                 float clip,
+                                 bool linear_before_reset)
+        : Op({X, H_t, WR, B}),
+          RNNCellBase(hidden_size, clip, activations, activations_alpha, activations_beta),
+          m_direction(direction),
+          m_linear_before_reset(linear_before_reset) {
+    constructor_validate_and_infer_types();
+}
+
+void op::GRUSequenceIE::validate_and_infer_types() {
+    element::Type arg_type = get_input_element_type(0);
+    PartialShape output_shape_0{PartialShape::dynamic(3)};
+    PartialShape output_shape_1{PartialShape::dynamic(2)};
+    if (get_input_partial_shape(0).is_static()) {
```

Comment:
Lets check that rank is static and greater that 2 and 0/1 dims are static and then get them for batch_size and seq_length